File: androidApp/src/main/java/com/kgmyshin/kmm/viewmodel/sample/android/ui/personlist/PersonListPage.kt
package com.kgmyshin.kmm.viewmodel.sample.android.ui.personlist

import androidx.compose.runtime.Composable
import androidx.compose.runtime.collectAsState
import androidx.compose.runtime.getValue
import com.kgmyshin.kmm.viewmodel.sample.ui.personlist.PersonListViewModelFactory
import com.kgmyshin.kmm.viewmodel.viewmodel.ktx.rememberViewModel

@Composable
fun PersonListPage(viewModelFactory: PersonListViewModelFactory) {
    val viewModel =
        rememberViewModel(viewModelFactory = viewModelFactory)
    val personList by viewModel.personListStateFlow.collectAsState()

    PersonListTemplate(personList = personList, onClickDelete = viewModel::onClickDelete)
}

File: iosApp/iosApp/PersonListPage.swift
import SwiftUI
import shared

struct PersonListPage: View {
    var viewModel: PersonListViewModel
    
    @ObservedObject var personList: FlowState<[PersonUiModel]>
    
    init() {
        let factory = koin.get(objCClass: PersonListViewModelFactory.self) as! PersonListViewModelFactory
        self.viewModel = ViewModelKt.createViewModel(viewModelFactory: factory) as! PersonListViewModel
        self.personList = self.viewModel.collectAsState(flow: self.viewModel.personListStateFlow)
    }
    
    var body: some View {
        PersonListTemplate(
            personList: self.personList.value ?? [], onClickDelete: { id in
                self.viewModel.onClickDelete(id: id)
            }
        ).onAppear(perform: {
            viewModel.onLaunched()
        })
        .onDisappear(perform: {
            viewModel.onDispose()
        })
    }
}
